accounts/abi/bind/backends/simulated.go

```diff
@@ -130,6 +134,65 @@ func (b *SimulatedBackend) rollback() {
 	b.pendingState, _ = state.New(b.pendingBlock.Root(), b.blockchain.StateCache(), nil)
 }
 
+// Reorg simulates a chain reorg with parameters depth and length.
+// f can be used to insert, reorder and exclude transactions.
+// f receives a slice of Transactions where each slice entry contains the
+// transactions that were contained in the orhpaned block.
+// The slice that is passed to f has length depth, and the one returned
+// must have at least length length.
+// Panics if length is not greater than depth.
+func (b *SimulatedBackend) Reorg(ctx context.Context, depth, length int64, f Reorderer) error {
+	if depth <= 0 {
+		panic("depth must be greater than 0")
+	}
+	// The chain inserter always chooses the most difficult chain, so a
+	// shorter chain would not trigger a reorg.
```

Comment:
That's not entirely true, a shorter chain can trigger a reorg if it's difficulty is higher, but I think we can ignore that for the simulated backend. We can change that if someone complains later.

Reply:
Yes right, i tried to change the difficulty of the inserted blocks by using `block.SetDifficulty` but then they fail to verify with:  
`inserting new chain: invalid difficulty: have 111111111111, want 131072`.  
Maybe the mocked hasher or the difficulty consensus is not happy then.  
  
I will change the comment to:  
```pre
// The chain inserter always chooses the most difficult chain.
// In the simulated case we expect that a longer chain is always
// more difficult and will therefore be picked in a reorg case.
// This check makes the function deterministic in the length<=depth case.
```